pageserver/src/tenant/timeline.rs

```diff
@@ -3542,6 +3647,8 @@ impl Timeline {
             })
             .collect::<Vec<_>>();
 
+        let deltas_to_compact_layers = deltas_to_compact.iter().collect_vec();
```

Comment:
Why

1. do we need a shallow copy?
2. don't you create the shallow copy using `Clone::clone`?

Reply:
I think this is some leftovers after merging the latest main branch and I can remove it now :)